packages/mattermost-redux/src/actions/channels.ts

```diff
@@ -525,7 +525,24 @@ export function fetchAllMyTeamsChannelsAndChannelMembers(): ActionFunc {
         const state = getState();
         const {currentUserId} = state.entities.users;
         let channels;
-        let channelsMembers;
+        let channelsMembers: ChannelMembership[] = [];
+        let allMembers = true;
+        let page = 0;
+        do {
+            try {
+                await Client4.getAllChannelsMembers(currentUserId, page, 100).then((data) => {
```

Comment:
@hmhealey I am not sure if we should do it like this. Take look and let me know.

Reply:
Can we consider to paginate via scrolling? 
Do we need for all users to be available?

Reply:
We need all the users available

Reply:
How soon do we expect the followup PR to be to improve the server API? I'd be worried that this has the potential to increase the number of heavy requests on page load for some of our power users that already have issues with page load. Looking at community, I have around 300 channels and DMs loaded across both teams, so that's at least a couple extra requests that we'll be waiting on. And then some people like Jason B who have potentially double that

Reply:
Do we really need a temporary fix for this then if the server changes will also be in before 7.0?

Reply:
I discussed with @Willyfrog that we might miss the 7.0 Deadline due to other priority changes we are working on.
I have removed the loop. Instead of calling API with 600 items per page, this will work for 95% of the user but fail for some people like Jason B.

Reply:
We could do the requests in batches, like in the loop you did before, and stop fetching after a certain number of pages have been fetched
e.g: 10 pages in batches of 60 would be 600
0/5

Reply:
Thanks! I like the idea of just doing one big page. That should end up being the same amount of load for server and client, but without having to make ~10 different connections to the server. Plus, it's bounded

Reply:
The max page size limit is controlled by the server of course. Otherwise the client can ask for 10000 items and crash the server. Currently the max limit is at 200 and I don't think bumping it to 600 makes a huge difference. We should move to streaming APIs as the permanent fix, but till then 200 should be fine.

Reply:
Oh I think I need to bring back the loop, Thanks @agnivade

Reply:
Yep, please loop with batch size of 200. Personally, I am not a big fan of this and I'd like to have the proper solution from the server side. I'd be okay with this remaining broken until we fix it properly. Making multiple requests just places additional load on the server.